src/api/spec/controllers/public_controller_spec.rb

```diff
@@ -1,14 +1,130 @@
 require 'rails_helper'
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the actions
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
 
 RSpec.describe PublicController, vcr: true do
-  let(:project) { create(:project, name: "public_controller_project") }
+  let(:project) { create(:project, name: "public_controller_project", title: "The Public Controller Project") }
   let(:package) { create(:package_with_file, name: "public_controller_package", project: project) }
 
   describe "GET #source_file" do
-    it "sends the requested file" do
+    before do
       get :source_file, project: project.name, package: package.name, filename: "somefile.txt"
-      expect(response).to have_http_status(:success)
-      expect(response.body).to eq(package.source_file("somefile.txt"))
     end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to eq(package.source_file("somefile.txt")) }
+  end
+
+  describe "GET #index" do
+    before do
+      get :index
+    end
+
+    it { is_expected.to respond_with(302) }
+    it { is_expected.to redirect_to(about_path) }
+  end
+
+  describe "GET #build" do
+    before do
+      get :build, project: project.name
+    end
+
+    it { is_expected.to respond_with(:success) }
+  end
+
+
+  describe "GET #configuration" do
+    before do
+      get :configuration_show
```

Comment:
What about the format of the response? (xml and json are expected) Does it worth to test the responses?